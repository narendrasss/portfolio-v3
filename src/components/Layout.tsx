import React from 'react';
import Helmet from 'react-helmet';
import styled from 'styled-components';
import favicon from '../images/favicon.ico';
import { useMetadata } from '../hooks';
import Flex, { FlexProps } from './base/Flex';

interface LayoutProps extends FlexProps {
  title?: string;
}

const Layout: React.FC<LayoutProps> = ({
  title = 'Narendra Syahrasyad',
  children,
  ...props
}) => {
  const { description } = useMetadata();
  return (
    <Flex
      as="main"
      height="100vh"
      width="100vw"
      flexDirection="column"
      p={1}
      {...props}
    >
      <Helmet
        title={title}
        meta={[{ name: 'description', content: description }]}
        link={[{ rel: 'shortcut icon', type: 'image/png', href: `${favicon}` }]}
      />
      {children}
    </Flex>
  );
};

export default Layout;
