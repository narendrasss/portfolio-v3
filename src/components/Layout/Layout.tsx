import React, { useState } from 'react';
import { useMetadata } from '../../hooks';
import { MainProps } from '../Main/Main';
import Nav from '../Nav/Nav';
import Content from '../Content/Content';

interface LayoutProps extends MainProps {
  title?: string;
}

const Layout: React.FC<LayoutProps> = ({
  title = 'Narendra Syahrasyad',
  ...rest
}) => {
  const { description, url, links } = useMetadata();
  const [isNavOpen, setIsNavOpen] = useState(false);
  const handleClick = () => setIsNavOpen(prev => !prev);

  const props = {
    description,
    links,
    title,
    url,
    onNavClick: handleClick,
  };

  if (isNavOpen) return <Nav data-testid="nav" {...props} {...rest} />;
  return <Content data-testid="content" {...props} {...rest} />;
};

export default Layout;
