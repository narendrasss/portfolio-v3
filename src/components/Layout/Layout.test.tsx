import React from 'react';
import { render, fireEvent, waitForElement } from 'react-testing-library';
import Layout from './Layout';

jest.mock('../../hooks');

describe('Layout', () => {
  const component = <Layout title="Narendra Syahrasyad" />;

  it('renders content on first mount', () => {
    const { queryByTestId } = render(component);
    expect(queryByTestId('content')).toBeTruthy();
    expect(queryByTestId('nav')).toBeNull();
  });

  it('renders nav when nav button is clicked', () => {
    const { queryByTestId, getByTestId } = render(component);
    fireEvent.click(getByTestId('nav-button'));
    expect(waitForElement(() => queryByTestId('nav'))).resolves.toBeTruthy();
  });
});
