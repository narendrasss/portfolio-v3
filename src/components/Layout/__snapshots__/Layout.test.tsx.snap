// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layout renders without crashing 1`] = `
<DocumentFragment>
  <main
    class="sc-bdVaJa sc-bwzfXH sc-gZMcBi bKgYbR"
    data-testid="content"
  >
    <header
      class="sc-bdVaJa sc-bwzfXH sc-iwsKbI dUmjld"
      width="100%"
    >
      <p
        class="sc-bdVaJa sc-htpNat jzCLQO"
        font-family="sans"
        font-size="0,1"
        font-weight="600"
      >
        narendras
        <span
          class="sc-bdVaJa sc-htpNat fYauba"
          font-family="sans"
          font-size="0,1"
          font-weight="300"
        >
          .io
        </span>
      </p>
      <nav
        class="sc-bdVaJa sc-dnqmqq hjWSCW"
        display="none,block"
      >
        <a
          class="sc-bdVaJa sc-htoDjs jssEWI sc-htpNat dRQZBK"
          color="black"
          font-family="sans"
          font-size="0,1"
          href="/projects"
        >
          projects
        </a>
        <a
          class="sc-bdVaJa sc-htoDjs jssEWI sc-htpNat dRQZBK"
          color="black"
          font-family="sans"
          font-size="0,1"
          href="/about"
        >
          about
        </a>
        <a
          class="sc-bdVaJa sc-htoDjs jssEWI sc-htpNat dRQZBK"
          color="black"
          font-family="sans"
          font-size="0,1"
          href="/blog"
        >
          blog
        </a>
        <a
          class="sc-bdVaJa sc-htoDjs jssEWI sc-htpNat dRQZBK"
          color="black"
          font-family="sans"
          font-size="0,1"
          href="/resume"
        >
          resume
        </a>
      </nav>
    </header>
    <button
      class="sc-bdVaJa sc-gqjmRU sc-VigVT bvdyGS sc-dnqmqq kXPwVz"
      data-testid="nav-button"
      display="block,none"
      type="button"
      width="4.5rem"
    >
      <div
        class="sc-bdVaJa sc-dnqmqq sc-fjdhpX hFTANy"
        width="4.5rem"
      />
      <div
        class="sc-bdVaJa sc-dnqmqq sc-fjdhpX cVrGbe"
        width="3.6rem"
      />
      <div
        class="sc-bdVaJa sc-dnqmqq sc-fjdhpX gTeBaA"
        width="2.25rem"
      />
    </button>
  </main>
</DocumentFragment>
`;
